src/security/Crypto.js

```diff
@@ -179,10 +226,50 @@ class Crypto {
      * Converts a key from a string to buffer.
      *
      * @param {string} key - base64Url encoded key
-     * @return {Buffer} key - key in Buffer form
+     * @return {Uint8Array} buffered key
      */
     static bufferKey(key) {
         return Crypto.wrapBuffer(base64Url.toBuffer(key));
     }
+
+    /**
+     * Converts an ECDSA signature from P1363 to DER format
+     *
+     * @param {ArrayBuffer} sig - P1363 signature
+     * @return {Uint8Array} DER signature
+     * @private
+     */
+    static _convertSigToDER(sig) {
+        const signature = Array.from(new Uint8Array(sig), (x) => ('00' + x.toString(16)).slice(-2)).join('');
+        let r = signature.substr(0, signature.length / 2);
+        let s = signature.substr(signature.length / 2);
+        r = r.replace(/^(00)+/, '');
+        s = s.replace(/^(00)+/, '');
+        if ((parseInt(r, 16) & '0x80') > 0) r = `00${r}`;
+        if ((parseInt(s, 16) & '0x80') > 0) s = `00${s}`;
+        const rString = `02${(r.length / 2).toString(16).padStart(2, '0')}${r}`;
+        const sString = `02${(s.length / 2).toString(16).padStart(2, '0')}${s}`;
+        const derSig = `30${((rString.length + sString.length) / 2).toString(16).padStart(2, '0')}${rString}${sString}`;
+        return new Uint8Array(derSig.match(/[\da-f]{2}/gi).map((h) => parseInt(h, 16)));
+    }
+
+    /**
+     * Converts an ECDSA signature from DER to P1363 format
```

Comment:
assuming that r and s are the same length

Reply:
not sure what you mean here, r and s are always the same length for P1363

Reply:
but the implicit assumption here is that the r and s inside the DER sig are also the same length (for it to be convertble). just would be nice to clarify in comments

Reply:
r and s in DER are not guaranteed to be the same length
https://stackoverflow.com/a/39575576
first remove the leading double zeros and then pad a 00 if the first bit is 1.